nrf_modem/doc/fault_handling.rst

```diff
@@ -0,0 +1,48 @@
+.. __fault_handling:
+
+Modem fault handling
+####################
+
+The application core and the modem core are separate cores on nRF9160.
+When the modem core on the nRF9160 crashes, it sends a fault signal to the application core.
+
+Detecting a fault
+*****************
+
+The Modem library implements a fault handling mechanism allowing the application to be notified when the modem core has crashed.
+The application can set a modem fault handler function during initialization using the Modem library initialization parameters.
+For a complete list of modem fault reasons, see :ref:`nrf_modem_fault`.
+
+In |NCS| the :ref:`nrf_modem_lib_readme` sets the modem fault handler during Modem library initialization.
+
+How to recover from a fault
+***************************
+
+When the modem has crashed, it can be re-initialized by the application.
+The application core does not need to be reset in order to recover the modem from a fault state.
+The application can re-initialize the modem by re-initializing the Modem library, by calling :c:func:`nrf_modem_shutdown` and :c:func:`nrf_modem_init`.
```

Comment:
```suggestion
When the modem has crashed, it can be reinitialized by the application.
The application core does not need to be reset to recover the modem from a fault state.
The application can reinitialize the modem by reinitializing the Modem library through calls to :c:func:`nrf_modem_shutdown` and :c:func:`nrf_modem_init` functions.
```